### DISCUSSION TITLE
Could not LIST the contents in s3 bucket as it is not treated as a directory

### DISCUSSION THREAD

**Discussion by ompushkara** [fclairamb/ftpserverlib#375] [General]
I tried listing the root directory which is the bucket using "/" and "." , none worked . It throws a 550 error stating that Cant change directory to / Not a directory.

Note : I am able to list the contents of the directory created under a bucket though.

**Comment by drakkan** [comment]
Hi,

without looking at your code is not possible to understand what is wrong. I think this is an error on your side, [SFTPGo](https://github.com/drakkan/sftpgo) does not have the reported issue and I think [ftpserver](https://github.com/fclairamb/ftpserver) works as well. They both use this library

**Reply by ompushkara** [reply]
Sure i am creating a new bucket in go using afero-s3 [afero-s3](https://github.com/fclairamb/afero-s3)

aferos3.NewFs(bucketname,session)

This creates the bucket inside the server . Now i use ftpclient to put a file inside the bucket . I still can login inside the bucket to view the file . But when i use LIST command on bucket assuming its "/" , i cannot view the contents and it throws a 550 error. Is there a way i can list the contents of the bucket . I can still create a new directory inside bucket , upload a file and do a LIST on the directory . That works pretty fine.

I think i just need a heads up might be on **how to list the contents of a S3 bucket** as i might be doing the incorrect way as you mentioned 

Note : The ftp client I use is lftp (https://lftp.yar.ru/lftp-man.html)

**Reply by drakkan** [reply]
> Sure i am creating a new bucket in go using afero-s3 [afero-s3](https://github.com/fclairamb/afero-s3)
> 
> aferos3.NewFs(bucketname,session)
> 
> This creates the bucket inside the server . Now i use ftpclient to put a file inside the bucket . I still can login inside the bucket to view the file . But when i use LIST command on bucket assuming its "/" , i cannot view the contents and it throws a 550 error. Is there a way i can list the contents of the bucket . I can still create a new directory inside bucket , upload a file and do a LIST on the directory . That works pretty fine.
> 
> I think i just need a heads up might be on **how to list the contents of a S3 bucket** as i might be doing the incorrect way as you mentioned
> 
> Note : The ftp client I use is lftp (https://lftp.yar.ru/lftp-man.html)

I have never used afero-s3 so I cannot help here sorry. Can you confirm that the same problem also occurs with [ftpserver](https://github.com/fclairamb/ftpserver)?

**Reply by ompushkara** [reply]
It doesn't work there as well I guess and thanks for pointing me in the right direction since there is an issue raised already for that : https://github.com/fclairamb/ftpserver/issues/801 

Hope they add the fix soon

**Reply by drakkan** [reply]
> It doesn't work there as well I guess and thanks for pointing me in the right direction since there is an issue raised already for that : [fclairamb/ftpserver#801](https://github.com/fclairamb/ftpserver/issues/801)
> 
> Hope they add the fix soon

Ah, I see. I don't like the solution in the linked issue. I think it should be fixed in afero-s3, but as said I've never used it so I'm not sure how it works

**Reply by ompushkara** [reply]
You may be right as I see a PR created there as well for this fix but not yet merged : https://github.com/fclairamb/afero-s3/pull/393/files . Hope it gets merged soon.
